amgut/test/test_auth_handlers.py

```diff
@@ -0,0 +1,133 @@
+from unittest import main
+
+from tornado.escape import url_escape
+
+from amgut.connections import ag_data
+from amgut.test.tornado_test_base import TestHandlerBase
+from amgut import text_locale
+
+
+class TestAuthBasehandler(TestHandlerBase):
+    def test_set_current_user(self):
+        # test if new cookie will be set
+        response = self.get('/auth/register/')
+        self.assertIn('Set-Cookie', response.headers.keys())
+
+        # test that cookie is re-used
+        self.mock_login('tst_ACJUJ')
+        response = self.get('/auth/register/')
+        self.assertNotIn('Set-Cookie', response.headers.keys())
+
+
+class TestAuthRegisterHandoutHandler(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/auth/register/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/auth/register/' % port)
+
+    def test_post(self):
+        # test reaction to invalid kit IDs
+        response = self.post('/auth/register/',
+                             {'kit_id': 'wrong_kit_id',
+                              'password': 'test',
+                              'email': 'e@mail',
+                              'email2': 'e@mail',
+                              'participantname': 'joe',
+                              'address': 'street',
+                              'city': 'metropolis',
+                              'state': 'XX',
+                              'zip': '12345',
+                              'country': 'Afghanistan'})
+        port = self.get_http_port()
+        self.assertEqual(
+            response.effective_url,
+            'http://localhost:%d/?loginerror=%s' %
+            (port, url_escape(text_locale['handlers']['INVALID_KITID'])))
+
+        unregistered_kits = ag_data.get_all_handout_kits()
+        if len(unregistered_kits) <= 0:
```

Comment:
Not sure - but I'm more concerned if a change in the order of the tests is going to fire up a stochastic failure. If we don't foresee this issue, what we can do is open an issue, put a TODO comment on the code with the link to the issue and try to fix later, so we don't loose track of that. I don't know from the top of my head how to revert this changes in the database, as that is specific to the code being executed, but I was told that in this repository, instead of reseting the full database in each test, the local changes are reverted